### ISSUE TITLE
Play sound when espanso is triggered

### ISSUE THREAD

**Issue by AucaCoyan** [federico-terzi/espanso#2135]
@killtux originally created this thread:

### Discussed in https://github.com/espanso/espanso/discussions/1170

<div type='discussions-op-text'>

<sup>Originally posted by **killtux** April  5, 2020</sup>
i would be nice to have an option for a sound notification the moment espanso is triggered. for example like "beeftext" for windows does. it should be simple and not too distracting.</div>

And also @greggjoseph1 also created in another thread a similar idea:

<div type='discussions-op-text'>

<sup>Originally posted by **greggjoseph1** December 13, 2022</sup>
Love espanso and can't live without it now!
I've just started a new match file which I'll populate over time with my common typos and misspellings to help me out in apps that don't have built in spellchecks. 
I normally have all espanso notifications disabled because I always know it's working when I invoke the trigger, but it would be cool to know that it's made a replacement of a word that I misspelled (matchs in that specific spellchecker.yml) that I may have done accidentally without realising. 
To implement this I envision a type of "show_notifications: true" that may optionally be set at the top of match files and only applies to matches made in that file's triggers. </div>

**Comment by smeech** [comment]
Current (Linux) workaround for individual triggers, depends on local audio player:
```yml
global_vars:
  - name: playsound
    type: shell  
    params:
      cmd: paplay $HOME/path/to/pull-back-325.ogg

matches:
  - trigger: :test
    replace: something{{playsound}}
```

**Comment by Luminus** [comment]
Why am I not surprised that @smeech already made something happen.

macOS has a local audio player called `afplay`, so swapping out `paplay` above for `afplay` works.

There does seem to be an appreciable delay, and the sound plays first before the replacement text is populated, which can be discombobulating.

**Comment by smeech** [comment]
> the sound plays first

Yes, Espanso evaluates all the variables prior to inserting the `replace:` values.

**Comment by Luminus** [comment]
> > the sound plays first
> 
> Yes, Espanso evaluates all the variables prior to inserting the `replace:` values.

That makes sense.

Hopefully, we can figure out a sustainable way to make this work. 

It's been fun tinkering with this, for sure.

**Comment by pitkling** [comment]
Here's another relevant discussion where this workaround came up a while ago (just to link this issue and that discussion, so everyone is kept in the loop on future progress): #1465.

**Comment by smeech** [comment]
Thank you for linking that in. Apologies for not attributing you as the originator of the workaround!

**Comment by pitkling** [comment]
Not at all, don't worry! I'm happy if it spreads and we eventually find a better workaround (or the feature gets implemented :)).

**Comment by mfichtner** [comment]
You can eliminate the delay by running the audio player in the background. The following works on macOS (but it should be just the same with `paplay` or any other player on Linux machines):

```yaml
global_vars:
  - name: tick
    type: shell
    params:
      cmd: "nohup afplay -v 0.25 /usr/local/lib/tick.aif </dev/null >/dev/null 2>&1 &"
```

The `&` operator backgrounds the process, and `nohup` detaches it from the invoking shell, so that it doesn't get killed prematurely.
